.evergreen/scripts/compile-unix.sh

```diff
@@ -82,8 +82,14 @@ configure_flags_append_if_not_null SRV "-DENABLE_SRV=${SRV}"
 configure_flags_append_if_not_null TRACING "-DENABLE_TRACING=${TRACING}"
 configure_flags_append_if_not_null ZLIB "-DENABLE_ZLIB=${ZLIB}"
 
+if [[ "${DEBUG:-}" == "ON" && "${RELEASE:-}" == "ON" ]]; then
```

Comment:
> We should consider refactoring the config to use only one or the other.

Consider removing `RELEASE` since it appears unused. Though it could be done in a future PR.

Reply:
`RELEASE` is used by the `release-compile` EVG task. It may be preferable to remove `DEBUG` instead and let all EVG tasks use Debug build type by default.